Sources/Foundation/JSONEncoder.swift

```diff
@@ -237,176 +237,405 @@ open class JSONEncoder {
     /// - returns: A new `Data` value containing the encoded JSON data.
     /// - throws: `EncodingError.invalidValue` if a non-conforming floating-point value is encountered during encoding, and the encoding strategy is `.throw`.
     /// - throws: An error if any value throws an error during encoding.
-    open func encode<T : Encodable>(_ value: T) throws -> Data {
-        let encoder = _JSONEncoder(options: self.options)
- 
-        guard let topLevel = try encoder.box_(value) else {
+    open func encode<T: Encodable>(_ value: T) throws -> Data {
+        let value: JSONValue = try encodeAsJSONValue(value)
+        let writer = JSONValue.Writer(options: self.outputFormatting)
+        let bytes = writer.writeValue(value)
+        
+        return Data(bytes)
```

Comment:
If `JSONValue.Writer` wrote directly into a `Data` instead of a `[UInt8]` then the `Data(bytes)` which is doing a copy of the `[UInt8]` could be eliminated and `return writer.writeValue(value)` used instead.

Reply:
I just run a small performance test
```
JSONValue to [UInt8] to Data | took: 0.48609s
JSONValue to Data            | took: 5.17270s
```

I guess for now we should stay with `[UInt8]`

Reply:
Yeah `[UInt8]` would be preferred I believe